sycl/include/sycl/queue.hpp

```diff
@@ -117,10 +118,85 @@ class __SYCL_EXPORT queue {
   queue(const async_handler &AsyncHandler, const property_list &PropList = {})
       : queue(default_selector(), AsyncHandler, PropList) {}
 
+#if __cplusplus >= 201703L
+  /// Constructs a SYCL queue instance using the device identified by the
+  /// device selector provided.
+  /// \param DeviceSelector is SYCL 2020 Device Selector, a simple callable that
+  /// takes a device and returns an int
+  /// \param AsyncHandler is a SYCL asynchronous exception handler.
+  /// \param PropList is a list of properties for queue construction.
+  template <
+      typename DeviceSelector,
+      typename = std::enable_if_t<
```

Comment:
Perhaps it is time to define this concept somewhere instead of copy-pasting everywhere?

Reply:
agreed. done.